framework/utils/TreeStructure.py

```diff
@@ -53,9 +53,9 @@ def parse(inFile,dType=None):
     @ In, dType, string, optional, type of processing to use (xml or getpot)
     @ Out, tree, InputTree, structured input
   """
-  #check type of file, make sure it's a file
-  if not isinstance(inFile,file):
-    inFile = file(inFile,'r')
+  #check type of file, and open if it is a string.
+  if type(inFile).__name__ in ['str','bytes','unicode']:
+    inFile = open(inFile,'r')
```

Comment:
I'm on the border on this bit of type checking.  This will error in a more ugly way later if `inFile` is neither a `file` nor a `string`, right?  This is probably better than what was there before, though.